### ISSUE TITLE
Unable to Upgrade PostGres Cluster

### ISSUE THREAD

**Issue by psavva** [CrunchyData/postgres-operator#2472]
Please ensure you do the following when reporting a bug:

**Provide a concise description of what the bug is:**
After upgrading the postgres operator using the Ansible Upgrade method, my Databases were not automatically upgraded.
I am also not able to upgrade the database via pgo upgrade command.

**Provide information about your environment.**
```
[root@FLATRO fhir]# pgo version
pgo client version 4.6.2
pgo-apiserver version 4.6.2
```

```
[root@FLATRO fhir]# uname -a
Linux FLATRO 3.10.0-1160.6.1.el7.x86_64 #1 SMP Tue Nov 17 13:59:11 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
[root@FLATRO fhir]# cat /etc/redhat-release
CentOS Linux release 7.9.2009 (Core)
```

```
[root@FLATRO fhir]# kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T16:51:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

```
[root@FLATRO postgres-operator]# pgo show cluster --all

cluster : ehealth4u-keycloak (crunchy-postgres-ha:centos7-12.3-4.4.0)
        pod : ehealth4u-keycloak-8445bbc5bf-6zgtn (Running) on flatro (1/1) (primary)
                pvc: ehealth4u-keycloak (2Gi)
        resources : CPU: 2 Memory: 2Gi
        deployment : ehealth4u-keycloak
        deployment : ehealth4u-keycloak-backrest-shared-repo
        deployment : ehealth4u-keycloak-pgadmin
        service : ehealth4u-keycloak - ClusterIP (10.96.52.233) - Ports (2022/TCP, 5432/TCP)
        service : ehealth4u-keycloak-pgadmin - ClusterIP (10.107.157.13) - Ports (5050/TCP)
        labels : deployment-name=ehealth4u-keycloak pg-pod-anti-affinity= pgo-version=4.4.0 pgouser=admin autofail=true crunchy-pgbadger=false crunchy-pgha-scope=ehealth4u-keycloak crunchy_collect=false name=ehealth4u-keycloak pg-cluster=ehealth4u-keycloak pgo-backrest=true workflowid=b2b97437-6418-4a2f-b083-ed1db2b2cc21 crunchy-pgadmin=true

cluster : ehealth4ufhir (crunchy-postgres-ha:centos7-12.3-4.4.0)
        pod : ehealth4ufhir-6c4564cfd4-l7jbw (Running) on flatro (1/1) (primary)
                pvc: ehealth4ufhir (10Gi)
        resources : CPU: 12 Memory: 16Gi
        deployment : ehealth4ufhir
        deployment : ehealth4ufhir-backrest-shared-repo
        service : ehealth4ufhir - ClusterIP (10.106.123.62) - Ports (2022/TCP, 5432/TCP)
        service : ehealth4ufhir-ext-np - ClusterIP (10.105.173.200) - Ports (2022/TCP, 5432/TCP)
        labels : crunchy-pgbadger=false crunchy-pgha-scope=ehealth4ufhir name=ehealth4ufhir pgo-backrest=true pgouser=admin autofail=true crunchy_collect=false deployment-name=ehealth4ufhir pg-cluster=ehealth4ufhir pg-pod-anti-affinity= pgo-version=4.4.0 workflowid=0db43f22-f084-4776-8ce7-d3b27fc95110

cluster : neha-api (crunchy-postgres-ha:centos8-13.2-4.6.2)
        pod : neha-api-5c5fd47875-vtzzk (Running) on davila (1/1) (primary)
                pvc: neha-api (2Gi)
        resources : CPU: 2 Memory: 4Gi
        deployment : neha-api
        deployment : neha-api-backrest-shared-repo
        service : neha-api - ClusterIP (10.106.187.171) - Ports (2022/TCP, 5432/TCP)
        service : neha-api-ext-np - ClusterIP (10.110.129.254) - Ports (2022/TCP, 5432/TCP)
        labels : pgouser=admin workflowid=15353dc3-6b6e-4e1a-8747-1dec754e4950 crunchy-pgha-scope=neha-api deployment-name=neha-api name=neha-api pg-cluster=neha-api pgo-version=4.6.2

cluster : neha-dgc-issuing (crunchy-postgres-ha:centos8-13.2-4.6.2)
        pod : neha-dgc-issuing-5f9cfb7597-6fl7c (Running) on roccas (1/1) (primary)
                pvc: neha-dgc-issuing (2Gi)
        resources : CPU: 6 Memory: 16
        deployment : neha-dgc-issuing
        deployment : neha-dgc-issuing-backrest-shared-repo
        service : neha-dgc-issuing - ClusterIP (10.103.105.23) - Ports (2022/TCP, 5432/TCP)
        labels : crunchy-pgha-scope=neha-dgc-issuing deployment-name=neha-dgc-issuing name=neha-dgc-issuing pg-cluster=neha-dgc-issuing pgo-version=4.6.2 pgouser=admin workflowid=d12a936c-8684-48d9-961f-58c5ed5e085d

cluster : neha-dgc-verifier (crunchy-postgres-ha:centos8-13.2-4.6.2)
        pod : neha-dgc-verifier-7c4c5889f7-qgkvg (Running) on davila (1/1) (primary)
                pvc: neha-dgc-verifier (2Gi)
        resources : CPU: 2 Memory: 4Gi
        deployment : neha-dgc-verifier
        deployment : neha-dgc-verifier-backrest-shared-repo
        service : neha-dgc-verifier - ClusterIP (10.96.100.178) - Ports (2022/TCP, 5432/TCP)
        labels : crunchy-pgha-scope=neha-dgc-verifier deployment-name=neha-dgc-verifier name=neha-dgc-verifier pg-cluster=neha-dgc-verifier pgo-version=4.6.2 pgouser=admin workflowid=8079e43d-1bf5-43f0-a36a-efe7bc83c1b4

```

You will notice new 2 version of centos and pg used:
Pre upgrade databases: **crunchy-postgres-ha:centos7-12.3-4.4.0)**
Post upgrade databases: **crunchy-postgres-ha:centos8-13.2-4.6.2**

**Provide clear steps to reproduce the bug.**
Upgraded from Version 4.4.0 to  4.6.2 using the Ansible installer
Notice that the databases are not upgraded.
Trying to upgrade manually:
```
[root@FLATRO postgres-operator]# pgo upgrade ehealth4ufhir
All active replicas will be scaled down and the primary database in this cluster will be stopped and recreated as part of this workflow!
WARNING: Are you sure? (yes/no): yes
Error: cannot upgrade from centos7-12.3-4.4.0 to centos8-13.2-4.6.2, upgrade task failed.
```

Also notice that since crunchydata repo no longer maintains 12.3, also having issues with the backrest image pull.
```
backrest-backup-ehealth4u-keycloak-fw4cj                  0/1     ImagePullBackOff   0          83m
backrest-backup-ehealth4ufhir-ht8x7                       0/1     ImagePullBackOff   0          13d
backrest-backup-neha-api-m4g6z                            0/1     Completed          0          89m
backrest-backup-neha-dgc-issuing-rgmls                    0/1     Completed          0          12d
backrest-backup-neha-dgc-verifier-9nphk                   0/1     Error              0          89m
backrest-backup-neha-dgc-verifier-9tbwd                   0/1     Error              0          89m
backrest-backup-neha-dgc-verifier-g8mlg                   0/1     Error              0          88m
backrest-backup-neha-dgc-verifier-hxxxg                   0/1     Completed          0          88m
ehealth4u-keycloak-8445bbc5bf-6zgtn                       1/1     Running            0          57d
ehealth4u-keycloak-backrest-shared-repo-b86898c9d-rzgjx   1/1     Running            0          170m
ehealth4u-keycloak-pgadmin-667b7fc86-5ldjx                1/1     Running            0          170m
ehealth4ufhir-6c4564cfd4-l7jbw                            1/1     Running            0          59d
ehealth4ufhir-backrest-shared-repo-6f657468f-4spqt        1/1     Running            0          170m
neha-api-5c5fd47875-vtzzk                                 1/1     Running            0          170m
neha-api-backrest-shared-repo-74b4666b57-6wx8v            1/1     Running            0          12d
neha-dgc-issuing-5f9cfb7597-6fl7c                         1/1     Running            0          12d
neha-dgc-issuing-backrest-shared-repo-64d7dccffb-fnnrk    1/1     Running            0          15d
neha-dgc-verifier-7c4c5889f7-qgkvg                        1/1     Running            0          170m
neha-dgc-verifier-backrest-shared-repo-cd8d9c6-nf9gt      1/1     Running            0          170m
postgres-operator-775568bc-68nx9                          4/4     Running            1          15d
```

**Attach applicable logs. Please do not attach screenshots showing logs unless you are unable to copy and paste the log data.**
[pgo logs.zip](https://github.com/CrunchyData/postgres-operator/files/6565761/pgo.logs.zip)

## Overview

Unable to upgrade my clusters after upgrading pgo operator

## Environment

Please provide the following details:

- Platform: (Kubernetes)
- Platform Version: (1.19.4)
- PGO Image Tag: (centos7-12.3-4.4.0 and centos8-13.2-4.6.2)
- Postgres Version (12 and 13)
- Storage: (rook-ceph)

**Comment by jkatz** [comment]
Did you run [`pgo upgrade`](https://access.crunchydata.com/documentation/postgres-operator/latest/upgrade/automatedupgrade/#postgresql-operator-automated-cluster-upgrade)?

**Comment by psavva** [comment]
> Did you run [`pgo upgrade`](https://access.crunchydata.com/documentation/postgres-operator/latest/upgrade/automatedupgrade/#postgresql-operator-automated-cluster-upgrade)?

Yes, I'm unable to upgrade the cluster with `pgo upgrade` nor can I find any way how to perform the upgrade now.

**Comment by jkatz** [comment]
I missed this in the original ticket, this seems relatively straightforward:

> Error: cannot upgrade from centos7-12.3-4.4.0 to centos8-13.2-4.6.2, upgrade task failed.

You can't do a Postgres major upgrade using `pgo upgrade`. You first need to update to 12.6, then perform the major upgrade using [`crunchy-upgrade`](https://access.crunchydata.com/documentation/crunchy-postgres-containers/latest/container-specifications/crunchy-upgrade/) or a similar method. A future PGO release will automate the major upgrade process.

**Comment by psavva** [comment]
Thank you so much @jkatz  
I'm sorry I missed that in the documentation.
I'm proceeding to upgrade to centos8-12.6-4.6.2 then to centos8-13.2-4.6.2

**Comment by psavva** [comment]
Hi @jkatz   I'm sorry to be a bother.

There is something I'm not getting about the upgrade path.
I've now upgraded to the latest postgres operator v4.7 using ansible.
This went OK.

I've managed to upgrade my 1 database to centos8-12.7-4.7.0
Now i'm trying to upgrade it to centos8-13.3-4.7.0, which is not working.

As per the Container Dependencies document: https://access.crunchydata.com/documentation/postgres-operator/4.7.0/configuration/compatibility/
I meet the dependency of version 12.7, why is it not upgrading further to 13.3 with pgo upgrade?
